src/storm/generator/JaniNextStateGenerator.cpp

```diff
@@ -1069,6 +1072,7 @@ std::vector<Choice<ValueType>> JaniNextStateGenerator<ValueType, StateType>::get
                                                                 automatonIndex, state, stateToIdCallback));
 
                     if (this->getOptions().isBuildChoiceOriginsSet()) {
+                        auto modelAutomatonIndex = model.getAutomatonIndex(parallelAutomata[automatonIndex].get().getName());
```

Comment:
This looks complicated, but fine.

Reply:
Yes, it is. Alternatively, one could store an index mapping between the `parallelAutomata` vector and the original indices in the jani model. However, my feeling is that the current solution is less error prone.